NobelLaureateDetails/Pages/Index.cshtml.cs

```diff
@@ -28,18 +27,31 @@ public JsonResult OnGet()
                     }
                 }
 
+
                 foreach (long id in filteredIds) {
-                    string laureateDetails = webClient.DownloadString("http://api.nobelprize.org/v1/laureate.json?id="+id);
+                    string laureateDetails = DownloadData("http://api.nobelprize.org/v1/laureate.json?id="+id);
 
-                    LaureateDetails array = LaureateDetails.FromJson(laureateDetails);
+                    LaureateDetails laureateDetailsArray = LaureateDetails.FromJson(laureateDetails);
```

Comment:
"array" is not a descriptive name for this variable, as the variable should describe the data it contains, not the type of data it contains.  As long as you're renaming it, drop the word "array" from the variable name entirely.  Just call it laureateDetails.